keps/sig-storage/1495-volume-populators/README.md

```diff
@@ -0,0 +1,483 @@
+    - Describe the mechanism: Also install the data-source-validator controller
+      and its associated VolumePopulator CRD.
+    - Will enabling / disabling the feature require downtime of the control
+      plane? no
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? no
+
+* **Does enabling the feature change any default behavior?**
+  Before this feature, kube API server would silently drop any PVC data source
+  that it didn't recognize, as if the client never populated the field. After
+  enabling this feature, the API server allows any object to be specified.
+  The data-source-validator controller will emit events for PVCs if the data
+  source is not recognized as one which is be handled by another controller.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Turning off the feature gate should causes the API server to go back to dropping
+  data sources for new PVCs. An admin would also want to remove any data populator
+  controllers, and the data-source-validator controller, and the VolumePopulator
+  CRD. Lastly, and unbound PVCs with data sources should be deleted, as they
+  will never bind.
+  
+  Alternatively, just disable the feature gate and remove the populator controllers,
+  but leave the data-source-validator controller and the VolumePopulator CRD,
+  and it will generate events on any PVCs which need to be deleted. Then it
+  could be left up to end users to delete their own PVCs which will never bind.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  PVC which previously had data sources that were invalided by the rollback
+  would be empty volumes. Users would need to delete and re-create those PVCs.
+
+* **Are there any tests for feature enablement/disablement?**
+  Not at this time.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  None.
+
+* **What specific metrics should inform a rollback?**
+  None.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and can't do that now.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  No.
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Simply look at the data source field of the PVCs. Non-empty data sources with
+  a Kind other than snapshot and PVC indicate this feature is in use. Also the
+  existence of any VolumePopulator.populator.storage.k8s.io CRs would indicate
+  that a populator is installed and could be used.
```

Comment:
How can I know that the associated data-source-validator controller is actually running?  Is there a lease or something to watch?